src/sage/rings/polynomial/polynomial_element.pyx

```diff
@@ -5523,6 +5526,18 @@ cdef class Polynomial(CommutativePolynomial):
         if tgt.ngens() > 1 and tgt._has_singular:
             g = flatten(self).gcd(flatten(other))
             return flatten.section()(g)
+
+        if algorithm not in ["pari", "generic"]:
```

Comment:
I'd prefer to use tuple instead of list. Though… I wonder if Cython can optimize this.